### DISCUSSION TITLE
retrieve non-reply tweet chain via API v2

### DISCUSSION THREAD

**Discussion by yxd0018** [tweepy/tweepy#2000] [Questions] [answered]
I know conversation_id is used to retrieve replies. That's not what I mean for threaded tweet chain. Instead I mean those published by the same author to overcome the char limit. Each tweet usually begins with sequence number. ex: https://twitter.com/fallacyalarm/status/1586871424995983361 -> 1586871419471732736 -> 1586871391680618496
With v2 API, I see the previous tweet is in includes.tweets with argument expansions': ['referenced_tweets.id',]. 
My question is how I can get all these tweets in chronicle order given a single id in the middle. 
Thanks.

**Comment by Harmon758** [comment, marked as answer]
Those Tweets are just replies as well.
You can use the conversation ID field of the Tweet, e.g.:
```pycon
>>> response = client.get_tweet(1586871424995983361, tweet_fields="conversation_id")
>>> response.data.conversation_id
1586871383673700353
```

**Reply by yxd0018** [reply]
Could you tell me if there is a way to return tweets in search_recent_tweets() in chronicle order? sort_order =relevancy returns the most recent one first. I have to reverse the order by myself now. Thanks

**Reply by Harmon758** [reply]
Please start new discussions if you have any unrelated questions.

**Comment by yxd0018** [comment]
Thanks. I trust too much into the printout of response object. Is there an existing utility function to pretty print out non-empty content of response or tweet object?

**Reply by Harmon758** [reply]
> Why am I not getting expansions or fields data with API v2 using [`Client`](https://docs.tweepy.org/en/latest/client.html#tweepy.Client)?
> 
> If you are simply printing the objects and looking at that output, the string representations of API v2 models/objects only include the default fields that are guaranteed to exist.
> 
> The objects themselves still include the relevant data, which you can access as attributes or by subscription.
> 
> There’s also a `data` attribute/key that provides the entire data dictionary.

https://tweepy.readthedocs.io/en/v4.12.0/faq.html#why-am-i-not-getting-expansions-or-fields-data-with-api-v2-using-client
